cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorer.java

```diff
@@ -22,43 +23,48 @@
 
     private static final String IGW_PREFIX = "igw-";
 
-    public boolean hasInternetGatewayOfSubnet(DescribeRouteTablesResult describeRouteTablesResult, String subnetId) {
-        Optional<RouteTable> routeTable = getRouteTableForSubnet(describeRouteTablesResult, subnetId);
-        return hasInternetGateway(routeTable, subnetId);
+    public boolean hasInternetGatewayOfSubnet(DescribeRouteTablesResult describeRouteTablesResult, String subnetId, String vpcId) {
+        Set<RouteTable> routeTables = getRouteTableForSubnet(describeRouteTablesResult, subnetId, vpcId);
+        return hasInternetGateway(routeTables, subnetId, vpcId);
     }
 
-    private Optional<RouteTable> getRouteTableForSubnet(DescribeRouteTablesResult describeRouteTablesResult, String subnetId) {
+    private Set<RouteTable> getRouteTableForSubnet(DescribeRouteTablesResult describeRouteTablesResult, String subnetId, String vpcId) {
         List<RouteTable> routeTables = describeRouteTablesResult.getRouteTables();
-        Optional<RouteTable> routeTable = Optional.empty();
+        Set<RouteTable> connectedRouteTables = new HashSet<>();
         for (RouteTable rt : routeTables) {
-            LOGGER.info("Searching the routeTable where routeTable is {} and the subnet is :{}", rt, subnetId);
-            for (RouteTableAssociation association : rt.getAssociations()) {
-                LOGGER.info("Searching the association where association is {} and the subnet is :{}", association, subnetId);
-                if (subnetId.equalsIgnoreCase(association.getSubnetId())) {
-                    LOGGER.info("Found the routeTable which is {} and the subnet is :{}", rt, subnetId);
-                    routeTable = Optional.ofNullable(rt);
-                    break;
+            if (rt.getVpcId().equalsIgnoreCase(vpcId)) {
```

Comment:
are there possible to exist route table without the specified vpc? If yes, could you please add a log message or any error if not found with the specific vpc

Reply:
If we consider the entire logic it queries all of the RouteTables for a region at once in one of the upper layers and this part iterates over all of them, I would not like to log those cases.
For example in the engesc that is attached to this ticket the customer has more than 130 route table it seems to be 129 unnecessary log lines to me.